From 9f59162ba2b2da500569f214967e1dfeaabff472 Mon Sep 17 00:00:00 2001
From: Yong Zhang <yong.zhang@windriver.com>
Date: Thu, 20 Jun 2013 16:24:43 +0800
Subject: [PATCH] dpa: DPAA dTSEC drops packets because RxFIFO resources not
 available and increments RPRD register count

Partly back port

commit b1cd921fb [FMD16 sync with NCSW GA 4.5 RC 3.]
commit 84ae241e2 [fmd: FMD17.2 integration]
from QorIQ-SDK-V1.3.2
to fix "RxFIFO resources not available and increments RPRD register
count" issue.

Signed-off-by: Yong Zhang <yong.zhang@windriver.com>
---
 .../net/dpa/NetCommSw/Peripherals/FM/Pcd/fm_cc.c   |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/drivers/net/dpa/NetCommSw/Peripherals/FM/Pcd/fm_cc.c b/drivers/net/dpa/NetCommSw/Peripherals/FM/Pcd/fm_cc.c
index ddbf845..45cc495 100644
--- a/drivers/net/dpa/NetCommSw/Peripherals/FM/Pcd/fm_cc.c
+++ b/drivers/net/dpa/NetCommSw/Peripherals/FM/Pcd/fm_cc.c
@@ -180,8 +180,6 @@ static t_Error FmPcdCcSetRequiredAction(t_Handle h_FmPcd, uint32_t requiredActio
         else
             h_AdTmp = PTR_MOVE(h_AdTmp, FM_PCD_CC_AD_ENTRY_SIZE);
 
-        if(p_CcNextEngineParamsTmp[i].shadowAction & requiredAction)
-            continue;
         switch(p_CcNextEngineParamsTmp[i].nextEngineParams.nextEngine)
         {
             case(e_FM_PCD_CC):
@@ -2659,8 +2657,7 @@ t_Error FmPcdCcReleaseModifiedDataStructure(t_Handle h_FmPcd, t_List *h_FmPcdOld
     {
         if(p_CcNewModifyAdditionalParams->p_AdTableNew)
             ((t_FmPcdCcNode *)(p_CcNewModifyAdditionalParams->h_CurrentNode))->h_AdTable    = p_CcNewModifyAdditionalParams->p_AdTableNew;
-        if(p_CcNewModifyAdditionalParams->numOfKeys)
-            ((t_FmPcdCcNode *)(p_CcNewModifyAdditionalParams->h_CurrentNode))->numOfKeys    = p_CcNewModifyAdditionalParams->numOfKeys;
+        ((t_FmPcdCcNode *)(p_CcNewModifyAdditionalParams->h_CurrentNode))->numOfKeys    = p_CcNewModifyAdditionalParams->numOfKeys;
         if(p_CcNewModifyAdditionalParams->p_KeysMatchTableNew)
             ((t_FmPcdCcNode *)(p_CcNewModifyAdditionalParams->h_CurrentNode))->h_KeysMatchTable    = p_CcNewModifyAdditionalParams->p_KeysMatchTableNew;
         memcpy(((t_FmPcdCcNode *)(p_CcNewModifyAdditionalParams->h_CurrentNode))->nextEngineAndRequiredAction, &p_CcNewModifyAdditionalParams->nextEngineAndRequiredAction, sizeof(t_FmPcdCcNextEngineAndRecuiredActionParams) * (FM_PCD_MAX_NUM_OF_KEYS));
-- 
1.7.5.4

